src/main/java/com/intrafind/log4shell/Log4shellUpdate.java

```diff
@@ -142,6 +161,22 @@ private static void delete(Path oldFile, List<Path> backups) {
     }
   }
 
+  private static boolean zipContainsLog4j(InputStream inputStream) throws IOException {
+    try (final ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+      for (ZipEntry entry = zipInputStream.getNextEntry(); entry != null; entry = zipInputStream.getNextEntry()) {
+        if ("org/apache/logging/log4j/core/lookup/JndiLookup.class".equals(entry.getName())) {
+          return true;
```

Comment:
What if more than one jar/class is found in one zip?

Reply:
That can only happen with a broken jar.

Also, if the jar in question is patched an the user misses one of the duplicate classes, the tool will still detect the remaining class.